src/providers/rust.rs

```diff
@@ -227,40 +236,29 @@ impl RustProvider {
         Ok(pkg)
     }
 
-    fn should_use_musl(app: &App, env: &Environment) -> Result<bool> {
+    fn should_use_musl(app: &App, env: &Environment) -> bool {
         if env.is_config_variable_truthy("NO_MUSL") {
-            return Ok(false);
+            return false;
         }
 
         if RustProvider::get_rust_toolchain_file(app).is_some() {
-            return Ok(false);
+            return false;
         }
 
         // Do not build for the musl target if using openssl
-        if RustProvider::uses_openssl(app)? {
-            return Ok(false);
+        if RustProvider::uses_dependency(app, "openssl") {
+            return false;
         }
 
-        Ok(true)
+        true
     }
 
-    fn uses_openssl(app: &App) -> Result<bool> {
-        // Check Cargo.toml
-        if let Some(toml_file) = RustProvider::parse_cargo_toml(app)? {
-            if toml_file.dependencies.contains_key("openssl")
-                || toml_file.dev_dependencies.contains_key("openssl")
-                || toml_file.build_dependencies.contains_key("openssl")
-            {
-                return Ok(true);
-            }
-        }
-
-        // Check Cargo.lock
-        if app.includes_file("Cargo.lock") && app.read_file("Cargo.lock")?.contains("openssl") {
-            return Ok(true);
-        }
-
-        Ok(false)
+    fn uses_dependency(app: &App, dep: &'static str) -> bool {
+        // Check if any of the files includes the dependency
+        ["Cargo.toml", "Cargo.lock"].iter().any(|file_name| {
+            app.includes_file(file_name)
+                && app.read_file(file_name).unwrap_or_default().contains(dep)
```

Comment:
i don't think we should use `unwrap_or_default` here, if we fail to read the file, something has gone wrong and the error should be handled.